### PR TITLE
[12.0][MIG] website_logo

### PR DESCRIPTION
**Pull request by eslAmer**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
This still doesn't include previous migrations commits. Please follow technical steps in https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-12.0

**Comment by eslAmer** [comment]
@pedrobaeza  ok what should i do now ?

File: website_logo/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+from . import models
```
Comment (nikul-serpentcs) [inline website_logo/__init__.py]: add new blank line

File: website_logo/models/__init__.py [hunk 2]
```diff
@@ -0,0 +1 @@
+from . import res_config,website
```
Comment (nikul-serpentcs) [inline website_logo/models/__init__.py]: same here

File: website_logo/views/res_config_view.xml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+<odoo>
+    <data>
```
Comment (nikul-serpentcs) [inline website_logo/views/res_config_view.xml]: remove ` <data>`

File: website_logo/views/res_config_view.xml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+        <record id="website_configuration_add_logo_id" model="ir.ui.view">
+            <field name="name">add logo to website configuration</field>
+            <field name="model">res.config.settings</field>
+            <field name="inherit_id" ref="website.res_config_settings_view_form"/>
+            <field name="arch" type="xml">
+                <xpath expr="//div[@class='content-group']" position="inside">
```
Comment (nikul-serpentcs) [inline website_logo/views/res_config_view.xml]: IMO here use `hasclass`

File: website_logo/views/res_config_view.xml [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+                    <div class="row">
+                        <label class="col-lg-3 o_light_label" for="website_logo"/>
+                        <field name="logo" widget="image" class="float-left oe_avatar"/>
+                    </div>
+                </xpath>
+            </field>
+        </record>
+    </data>
+</odoo>
```
Comment (nikul-serpentcs) [inline website_logo/views/res_config_view.xml]: add a new blank line

File: website_logo/views/website_templates.xml [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+     <template id="add_logo_to_website" inherit_id="website.layout_logo_show" customize_show="True" name="Show Logo">
+        <xpath expr="//header//a[hasclass('navbar-brand')]" position="replace">
+            <a href="/" class="navbar-brand logo">
+                <span t-field="website.logo" t-options="{'widget': 'image'}" role="img"
+                      t-att-aria-label="'Logo of %s' % website.name" t-att-title="website.name"/>
+            </a>
+        </xpath>
+    </template>
+    </data>
+</odoo>
```
Comment (nikul-serpentcs) [inline website_logo/views/website_templates.xml]: add a new blank line

**Review by nikul-serpentcs — COMMENTED** [review]
Improve Code

**Comment by eslAmer** [comment]
@nikul-serpentcs  thanks in advance it's done , for the next times how do i find my code issue ! i mean if there's any error log or some kind of that ?

**Comment by nikul-serpentcs** [comment]
> for the next times how do i find my code issue ! i mean if there's any error log or some kind of that ?

@eslAmer In Travis

File: website_logo/models/res_config.py [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+# Copyright 2016 Tecnativa - Antonio Espinosa
+# Copyright 2017 Tecnativa - David Vidal
+# Copyright 2018 eslaAmer - Eslam Amer
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+
+class Config(models.TransientModel):
+    _inherit = 'res.config.settings'
+    logo = fields.Binary(related='website_id.logo',
+                                 readonly=False)
```
Comment (simahawk) [inline website_logo/models/res_config.py:12]:
```py
logo = fields.Binary(
    related='website_id.logo',
    readonly=False
)
```

File: website_logo/models/website.py [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+# Copyright 2015 Agile Business Group - Lorenzo Battistini
+# Copyright 2016 Tecnativa - Antonio Espinosa
+# Copyright 2017 Tecnativa - David Vidal
+# Copyright 2018 eslaAmer - Eslam Amer
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+
+class Website(models.Model):
+    _inherit = 'website'
+    logo = fields.Binary()
```
Comment (simahawk) [inline website_logo/models/website.py:11]: empty line above `logo`

File: website_logo/views/res_config_view.xml [hunk 9]
```diff
@@ -0,0 +1,15 @@
+<odoo>
+        <record id="website_configuration_add_logo_id" model="ir.ui.view">
```
Comment (simahawk) [inline website_logo/views/res_config_view.xml:2]: s/website_configuration_add_logo_id/website_configuration_add_logo

File: website_logo/views/res_config_view.xml [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+            <field name="name">add logo to website configuration</field>
+            <field name="model">res.config.settings</field>
+            <field name="inherit_id" ref="website.res_config_settings_view_form"/>
+            <field name="arch" type="xml">
+                <xpath expr="//div[hasclass('content-group')]" position="inside">
+                    <div class="row">
```
Comment (simahawk) [inline website_logo/views/res_config_view.xml:8]: I tested on runbot. This view breaks the settings panel. Fix it like this:
```xml
<label class="o_form_label col-lg-3 o_light_label" for="logo"/>
```

File: website_logo/README.rst [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+
+Load a logo image to be used on website only. This allows to use an internal
+company logo (for reports) and a different website logo.
+
+Configuration
+=============
+
+To configure this module, you need to:
+
+* Go to 'Website'.
```
Comment (simahawk) [inline website_logo/README.rst:17]: Go to "Website -> Configuration -> Settings"

**Review by simahawk — CHANGES_REQUESTED** [review]
Additionally I've just spotted an issue (not related to the migration):

image

In the website HEAD we still use the company logo, which is really wrong (search engines and social platforms previews, etc)

**Pull request closed without merging**